x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/MlMemoryAutoscalingDecider.java

```diff
@@ -346,6 +347,7 @@ private long maxMemoryBytes(MlAutoscalingContext mlContext) {
                 // Memory SHOULD be recently refreshed, so in our current state, we should at least have an idea of the memory used
                 .mapToLong(t -> {
                     Long mem = getAnomalyMemoryRequirement(t);
+                    logger.warn("unexpected null for anomaly detection memory requirement for [{}]", MlTasks.jobId(t.getId()));
```

Comment:
Shouldn't this log message be surrounded by a check that `mem` is null?

I'm not sure it is safe to insert an `null` element into a LongStream and then call `max()` on it. I tried a something similar in JShell and it threw an NPE:

```
jshell>  List<Long> ll = Arrays.asList(1L, 3L, null, 4L);
jshell> ll.stream().mapToLong(l -> l).max()

|  Exception java.lang.NullPointerException: Cannot invoke "java.lang.Long.longValue()" because "<parameter1>" is null
|        at lambda$do_it$$0 (#7:1)
|        at ReferencePipeline$5$1.accept (ReferencePipeline.java:231)
|        at Spliterators$ArraySpliterator.forEachRemaining (Spliterators.java:992)
|        at AbstractPipeline.copyInto (AbstractPipeline.java:509)
|        at AbstractPipeline.wrapAndCopyInto (AbstractPipeline.java:499)
|        at ReduceOps$ReduceOp.evaluateSequential (ReduceOps.java:921)
|        at AbstractPipeline.evaluate (AbstractPipeline.java:234)
|        at LongPipeline.reduce (LongPipeline.java:503)
|        at LongPipeline.max (LongPipeline.java:466)
|        at (#7:1)
```

Reply:
Ugh, of course. 🤦